### ISSUE TITLE
ansible_managed from ansible.cfg not available as a template var

### ISSUE THREAD

**Issue by 1adam** [ansible/ansible#11317]
### Issue Type:

Bug Report
### Ansible Version:

`ansible 2.0.0 (devel 4d5a98d8d9) last updated 2015/06/18 10:55:47 (GMT -400)
  lib/ansible/modules/core: (detached HEAD 9acc7c402f) last updated 2015/06/17 14:22:00 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 2f967a949f) last updated 2015/06/17 14:22:01 (GMT -400)
  v1/ansible/modules/core: (detached HEAD f8d8af17cd) last updated 2015/06/17 14:22:26 (GMT -400)
  v1/ansible/modules/extras: (detached HEAD 495ad450e5) last updated 2015/06/17 14:22:49 (GMT -400)
  configured module search path = None`
### Ansible Configuration:

/etc/ansible/ansible.cfg is the default one
### Environment:

Ubuntu 14.04.1 LTS
### Summary:

The sole action in a playbook is to instantiate a template on the server.  The template contains only `"# {{ansible_managed}}"`.  Instead of placing the template file as expected, Ansible errors out, complaining that "ansible_managed" is undefined.
### Steps To Reproduce:

Ensure that you have 'ansible_managed' set in your /etc/ansible/ansible.cfg, and then simply try to parse a template file that contains the string "{{ansible_managed}}".
### Expected Results:

Template file is parsed, and "{{ ansible_managed }}" is replaced appropriately.
### Actual Results:

`fatal: [my.server.com]: FAILED! => {"msg": "UndefinedError: 'ansible_managed' is undefined", "failed": true, "changed": false}`

**Comment by askulkarni2** [comment]
I see this issue as well. I have an ANSIBLE_CFG environment variable set.

**Comment by sigio** [comment]
Running from git checkout here... when running from 'devel' I see this issue... when running from 'v1_last' it works as expected.

**Comment by abadger** [comment]
Looks like the special template variables haven't been implemented for v2.  ansible_managed, template_uid, etc.

**Comment by jimi-c** [comment]
# Closing This Ticket

Hi!

We believe the above commit should resolve this problem for you. This will also be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!

**Comment by sigio** [comment]
Reopen please, comparing 1.9.6 and 2.0.2.0 (both from pip)

$ cat plays/test.yml

```
# play to test templates and ansible_managed
---
- hosts: localhost
  vars:
    - testvalue: '{{lookup("template", "testtemplate.j2") }}'
  tasks:
    - name: debug ansible_managed
      debug: var=testvalue

$ cat plays/testtemplate.j2
---
# {{ ansible_managed }}
```

(ansible1.9)$ ansible-playbook plays/test.yml 

```
PLAY [localhost] ************************************************************** 

GATHERING FACTS *************************************************************** 
ok: [localhost]

TASK: [debug ansible_managed] ************************************************* 
ok: [localhost] => {
    "var": {
        "testvalue": "# Ansible managed: /opt/ansible/plays/testtemplate.j2, please do not edit by hand\n"
    }
}

PLAY RECAP ******************************************************************** 
localhost                  : ok=2    changed=0    unreachable=0    failed=0    
```

(ansible-2.0)$ ansible-playbook plays/test.yml 

```
PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [debug ansible_managed] ***************************************************
ok: [localhost] => {
    "testvalue": "VARIABLE IS NOT DEFINED!"
}

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0   
```

**Comment by memelet** [comment]
How can this be closed? This is major break. And still in 2.1.

**Comment by jimi-c** [comment]
This variable is (and had been in 1.9) defined when using the template module. It was not generally available as a variable.

**Comment by sigio** [comment]
https://github.com/ansible/ansible/pull/23408

**Comment by sedlund** [comment]
im confused about the status of this.  the issue is closed, it looks like something merged to fix it, but it does not appear to be fixed in 2.4.2.0.

**Comment by abadger** [comment]
This issue was fixed a long time ago (it was opened in the transition from Ansible 1.x to 2.x).  If the feature isn't working for you, open a new bug report with a case that reproduces the problem.  We often do not look at closed bug reports at all.
